library/Rules/CreditCardVisa.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of Respect/Validation.
+ *
+ * (c) Alexandre Gomes Gaigalas <alexandre@gaigalas.net>
+ *
+ * For the full copyright and license information, please view the "LICENSE.md"
+ * file that was distributed with this source code.
+ */
+
+namespace Respect\Validation\Rules;
+
+class CreditCardVisa extends CreditCard
+{
+
+    public function validate($input)
+    {
+        $input = preg_replace('([^0-9])', '', $input);
```

Comment:
A credit card like `This is a credit card: 1234 1234 1234 1234` will pass.

The best option would be to define the separator in the constructor and just replace the separator for `''`.